### PR TITLE
Bring in more rule configurations from `@foxglove/studio`

### PR DESCRIPTION
**Pull request by jtbandes**
Graduate these changes to the shared config.

### CODE REVIEW

File: configs/base.js [hunk 1]
```diff
@@ -29,8 +29,11 @@ module.exports = {
       },
     ],
 
+    "filenames/match-exported": "error",
```
Comment (amacneil) [inline configs/base.js:32]: I tested this rule out in the website repo, there are a couple of places we will need to ignore it (e.g. next.js requires files named `_app.tsx` and `404.tsx` which aren't valid function export names). But other than that seems fine.

I assume you checked the perf implications of it?

**Review by amacneil — APPROVED** [review]

File: configs/base.js (hunk 1 above)
Comment (jtbandes) [reply configs/base.js:32]: Yeah, didn't see it in the timing output of any lint runs in studio

Comment (jtbandes) [reply configs/base.js:32]: `promise-function-async` is the slowest in studio. I can still run `yarn lint` in 50 sec.

```
Rule                                          | Time (ms) | Relative
:---------------------------------------------|----------:|--------:
@typescript-eslint/promise-function-async     |  6097.230 |    35.6%
import/order                                  |  2961.612 |    17.3%
@typescript-eslint/no-floating-promises       |  2115.628 |    12.3%
@typescript-eslint/no-unused-vars             |   660.667 |     3.9%
@typescript-eslint/unbound-method             |   563.162 |     3.3%
@typescript-eslint/strict-boolean-expressions |   491.951 |     2.9%
react/no-deprecated                           |   430.047 |     2.5%
react/no-unused-prop-types                    |   356.396 |     2.1%
react/no-direct-mutation-state                |   345.526 |     2.0%
✔ Lint done...
yarn lint  50.11s user 3.10s system 145% cpu 36.662 total
```

Comment (jtbandes) [reply configs/base.js:32]: btw, ESLint still doesn't run rules in parallel although that is an active area of research: https://github.com/eslint/rfcs/pull/42

Comment (amacneil) [reply configs/base.js:32]: sweet

**Pull request merged**